dowhy/causal_refuters/evalue_sensitivity_analyzer.py

```diff
@@ -0,0 +1,301 @@
+)
+from dowhy.causal_estimators.regression_discontinuity_estimator import (
+    RegressionDiscontinuityEstimator,
+)
+from dowhy.causal_estimators.two_stage_regression_estimator import (
+    TwoStageRegressionEstimator,
+)
+
+
+class EValueSensitivityAnalyzer:
+    """
+    This class computes Ding & VanderWeele's E-value for unmeasured confounding. The E-value is the minimum
+    strength of association on the risk ratio scale that an unmeasured confounder would need to have with
+    both the treatment and the outcome, conditional on the measured covariates, to fully explain away a specific
+    treatment-outcome association.
+
+    See: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4820664/ and
+    https://dash.harvard.edu/bitstream/handle/1/36874927/EValue_FinalSubmission.pdf.
+    The implementation is based on the R package: https://github.com/cran/EValue.
+
+    :param estimate: CausalEstimate
+    :param outcome_var: Outcome variable values
+    :param true: A number to which to shift the observed estimate to. Defaults to 1 for ratio measures (RR, OR, HR)
+    and 0 for additive measures (OLS, MD). (Default = 0)
+    :param delta: The contrast of interest in the treatment/exposure, for estimator of class LinearRegressionEstimator.
+    (Default = 1)
+    """
+
+    def __init__(
+        self, estimate: CausalEstimate, outcome_var: pd.Series, true=0, delta=1
+    ):
+
+        self.estimate = estimate
+        self.true = true
+        self.delta = delta
+        self.logger = logging.getLogger(__name__)
+        self.sd_outcome = np.std(outcome_var)
+        self.stats = None
+
+        outcome_values = outcome_var.astype(int).unique()
+        if all([v in [0, 1] for v in outcome_values]):
+            raise NotImplementedError(
+                "E-Value sensitivity analysis is not currently implemented for binary outcomes"
+            )
+        if isinstance(
+            self.estimate.estimator,
+            (
+                Causalml,
+                Econml,
+                InstrumentalVariableEstimator,
+                RegressionDiscontinuityEstimator,
+                TwoStageRegressionEstimator,
+            ),
+        ):
+            raise NotImplementedError(
+                "E-Value sensitivity analysis is not currently implemented for this estimator"
+            )
+
+    def check_sensitivity(self):
+        """
+        Function to compute E-value for point estimate and confidence limits. The estimate and
+        confidence limits are converted to the risk ratio scale before the E-value is calculated.
+        """
+
+        est = self.estimate.value
+        se = self.estimate.get_standard_error()
+        if isinstance(se, np.ndarray):
+            se = se[0]
+
+        if isinstance(self.estimate.estimator, LinearRegressionEstimator):
+            self._evalue_OLS(est, se, self.sd_outcome, self.delta, self.true)
+        else:
+            est = self._to_md(est, self.sd_outcome)
```

Comment:
would `to_smd` be a more descriptive name since we conver the quantity to standardized mean difference?